src/events/rawGuildMemberUpdate.ts

```diff
@@ -1,26 +1,48 @@
 import { WSGuildMemberUpdate, AuditLogResult } from '../lib/types/DiscordAPI';
-import { Event, EventStore } from 'klasa';
+import { Event, EventStore, KlasaGuild } from 'klasa';
 import { GuildSettings } from '../lib/types/settings/GuildSettings';
 import { api } from '../lib/util/Models/Api';
+import { MessageEmbed } from 'discord.js';
+import { floatPromise, getDisplayAvatar } from '../lib/util/util';
+import { Events } from '../lib/types/Enums';
+import { MessageLogsEnum } from '../lib/util/constants';
 
 export default class extends Event {
 
 	public constructor(store: EventStore, file: string[], directory: string) {
 		super(store, file, directory, { name: 'GUILD_MEMBER_UPDATE', emitter: store.client.ws });
 	}
 
-	public async run(data: WSGuildMemberUpdate): Promise<void> {
+	public run(data: WSGuildMemberUpdate) {
 		const guild = this.client.guilds.get(data.guild_id);
-		if (!guild) return;
+		if (typeof guild === 'undefined') return;
 
-		guild.nicknames.set(data.user.id, data.nick);
-		this.client.userTags.create(data.user);
-		const member = await guild.members.fetch(data.user.id).catch(() => null);
-		if (!member) return;
-		// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
-		// @ts-ignore 2339
-		member._patch(data);
+		this.handleNicknameChange(guild, data);
+		floatPromise(this, this.handleRoleSets(guild, data));
+	}
+
+	private handleNicknameChange(guild: KlasaGuild, data: WSGuildMemberUpdate) {
+		// Get the current nickname, compare them both, if they are different, it changed
+		const previous = guild.nicknames.get(data.user.id);
+		const next = data.nick || null;
+
+		// Get the previous nickname
+		guild.nicknames.set(data.user.id, next);
+
+		// If the previous was unset or it's the same as the next one, skip
+		if (typeof previous === 'undefined' || previous === next) return;
 
+		if (guild.settings.get(GuildSettings.Events.MemberNicknameUpdate)) {
+			this.client.emit(Events.GuildMessageLog, MessageLogsEnum.Member, guild, () => new MessageEmbed()
+				.setColor(0xDCE775)
```

Comment:
Should probably be in an enum of sorts

Reply:
Related to #491, it's pending to do and will be done in another Pull Request.